packages/cli/test/controller.js

```diff
@@ -25,80 +30,208 @@ describe('generator-loopback4:controller', tests);
 
 describe('lb4 controller', () => {
   it('does not run without package.json', () => {
-    helpers
+    let tmpDir;
+    return helpers
       .run(generator)
+      .inTmpDir(dir => {
+        tmpDir = dir;
+        utils.setupTmpDir(tmpDir);
+      })
       .withPrompts(withInputProps)
       .then(() => {
         assert.noFile(withInputName);
       });
   });
   it('does not run without the loopback keyword', () => {
     let tmpDir;
-    helpers
+    return helpers
       .run(generator)
       .inTmpDir(dir => {
         tmpDir = dir;
-        fs.writeFileSync(
-          path.join(tmpDir, 'package.json'),
-          JSON.stringify({
-            keywords: ['foobar'],
-          })
-        );
+        utils.setupTmpDir(tmpDir);
       })
       .withPrompts(withInputProps)
       .then(() => {
         assert.noFile(withInputName);
       });
   });
 
-  describe('with input', () => {
-    let tmpDir;
-    before(() => {
+  describe('basic', () => {
+    describe('with input', () => {
+      let tmpDir;
+      before(() => {
+        return helpers
+          .run(generator)
+          .inTmpDir(dir => {
+            tmpDir = dir;
+            utils.setupTmpDir(dir);
+          })
+          .withPrompts(withInputProps)
+          .toPromise();
```

Comment:
This helper "before" hook seems to be duplicated at least in two places. Could you please extract it to a top-level helper shared by all `describe` suites?